source/frontend/src/features/leases/add/AddLeaseContainer.tsx

```diff
@@ -1,97 +1,106 @@
+import MapSideBarLayout from 'features/mapSideBar/layout/MapSideBarLayout';
+import SidebarFooter from 'features/properties/map/shared/SidebarFooter';
 import { FormikProps } from 'formik';
 import { Api_Lease } from 'models/api/Lease';
 import * as React from 'react';
-import { useContext, useRef, useState } from 'react';
-import { useHistory } from 'react-router';
+import { useMemo, useState } from 'react';
+import { useEffect, useRef } from 'react';
+import { useHistory } from 'react-router-dom';
 
-import { LeasePageNames, leasePages } from '../detail/LeaseContainer';
 import { useAddLease } from '../hooks/useAddLease';
-import { FormLease } from '../models';
+import { LeaseFormModel } from '../models';
 import AddLeaseForm from './AddLeaseForm';
 
-export interface IAddLeaseContainerProps {}
+export interface IAddLeaseContainerProps {
+  onClose: () => void;
+}
 
 export const AddLeaseContainer: React.FunctionComponent<
   React.PropsWithChildren<IAddLeaseContainerProps>
 > = props => {
-  const leasePage = leasePages.get(LeasePageNames.DETAILS);
-  const { setTrayPage } = useContext(SidebarStateContext);
-  const { setState } = useContext(MapStateContext);
-  const onClickManagement = () => setTrayPage(SidebarContextType.LEASE);
   const history = useHistory();
+  const formikRef = useRef<FormikProps<LeaseFormModel>>(null);
+  const { setState } = React.useContext(MapStateContext);
+  const initialForm = useMemo<LeaseFormModel>(() => {
+    return new LeaseFormModel();
+  }, []);
   const { addLease } = useAddLease();
   const [addLeaseParams, setAddLeaseParams] = useState<
     { lease: Api_Lease; userOverride?: string } | undefined
   >();
-  const formikRef = useRef<FormikProps<FormLease>>(null);
 
-  if (!leasePage) {
-    throw Error('The requested lease page does not exist');
-  }
-  const onSubmit = async (lease: Api_Lease) => {
-    const leaseResponse = await addLease(lease, (userOverrideMessage?: string) =>
-      setAddLeaseParams({ lease: lease, userOverride: userOverrideMessage }),
-    );
-    if (!!leaseResponse?.id) {
-      history.push(`/lease/${leaseResponse?.id}`);
-    }
-  };
+  const { search } = useMapSearch();
 
-  // when leaving this page, reset any selected lease properties to ensure that future uses of this page do not include previously selected values.
-  React.useEffect(() => {
+  useEffect(() => {
     return () => {
       setState({
         type: MapStateActionTypes.SELECTED_LEASE_PROPERTY,
         selectedLeaseProperty: null,
       });
     };
-  }, [setState]);
+  }, [initialForm, setState]);
+
+  const saveLeaseFile = async (leaseFile: Api_Lease) => {
+    const response = await addLease(leaseFile, (userOverrideMessage?: string) =>
+      setAddLeaseParams({ lease: leaseFile, userOverride: userOverrideMessage }),
+    );
+    formikRef.current?.setSubmitting(false);
+    if (!!response?.id) {
+      formikRef.current?.resetForm();
+      await search();
```

Comment:
Is this to make sure that we include any newly added properties on the map after the lease is saved?

Reply:
That is correct. Similar behaviour as the research and acquisition adds